### PR TITLE
L4D2 Server

### PR DESCRIPTION
**Pull request by lucaspenney**
Also changed steamcache references to lancachenet (name change)

### CODE REVIEW

**Comment by jesseryoung** [comment]
To get networking functioning on the server could you add the following to your PR:
In `docker-compose.prod.yml`
```yml
services:
  l4d2:
    ports:
      - "69.162.82.212:27015:27015/udp"
    networks:
      - l4d2-network
      - game-network
...
networks:
...
  l4d2-network:
    driver_opts:
      com.docker.network.bridge.name: bridge-l4d2
      com.docker.network.bridge.enable_ip_masquerade: "false"
    ipam:
      config:
      - subnet: 10.4.0.0/16
        ip_range: 10.4.1.0/24  
```

And add the following line to the end of `setup_networking.sh`
```sh
iptables -t nat -A POSTROUTING -s 10.4.0.0/16 ! -o bridge-l4d2 -j SNAT --to-source 69.162.82.212
```

**Comment by jesseryoung** [comment]
Add the following to the server.cfg to have people who connect use the fast download server for maps: 
```sv_downloadurl "http://69.162.82.212:38929/l4d2/"```

**Comment by lucaspenney** [comment]
L4D2 doesn't let you download mods from server, it's too cool for that.

File: docker-compose.yml [hunk 1]
```diff
@@ -2,55 +2,65 @@ version: "2"
...
-    depends_on: 
+    depends_on:
       - dns
       - cache
     volumes:
       - pz_db:/home/steam/Zomboid/db
       - pz_save:/home/steam/Zomboid/Saves
 
+  l4d2:
+    image: team-brh/l4d2:latest
```
Comment (jesseryoung) [inline docker-compose.yml:39]: Found out why I couldn't get it running locally. The DNS and Cache servers are only running on the `game-network` and you missed this in the compose entry.
```suggestion
    image: team-brh/l4d2:latest
    networks:
      - game-network    
```

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
       - "16261:16261"
       - "16261:16261/udp"
-      - "16262-16286:16262-16286"
\ No newline at end of file
+      - "16262-16286:16262-16286"
+  l4d2:
+    environment:
+      - IP_ADDRESS=0.0.0.0
+    ports:
+      - "27015:27015/udp"
```
Comment (jesseryoung) [inline docker-compose.dev.yml:22]:
```suggestion
      - "27015:27015/udp"
      - "27017:27015/udp"
```

**Pull request merged**